common/config/ringpop.go

```diff
@@ -295,6 +300,84 @@ func (provider *dnsProvider) Hosts() ([]string, error) {
 	return results, nil
 }
 
+type dnsSRVProvider struct {
+	UnresolvedHosts []string
+	Resolver        dnsHostResolver
+	Logger          log.Logger
+}
+
+func newDNSSRVProvider(
+	hosts []string,
+	resolver dnsHostResolver,
+	logger log.Logger,
+) *dnsSRVProvider {
+
+	set := map[string]struct{}{}
```

Comment:
What is happening here? Can we just pass `hosts` slice to `UreslovedHosts`?

Reply:
Ah yeah, I was confused by this as well but it seems the purpose is to remove any duplicates. See: https://github.com/uber/cadence/pull/1941#discussion_r290788721